tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -420,23 +421,43 @@ def read_model(path_to_xml):
     mean_data = deepcopy(graph.graph['mf']) if 'mf' in graph.graph else None
     input_names = deepcopy(graph.graph['input_names']) if 'input_names' in graph.graph else []
 
-    prepare_emit_ir(graph=graph,
-                    data_type=graph.graph['cmd_params'].data_type,
-                    output_dir=argv.output_dir,
-                    output_model_name=argv.model_name,
-                    mean_data=mean_data,
-                    input_names=input_names,
-                    meta_info=non_default_params,
-                    use_temporary_path=True)
-
-    # This graph cleanup is required to avoid double memory consumption
-    graph.clear()
-
     output_dir = argv.output_dir if argv.output_dir != '.' else os.getcwd()
     orig_model_name = os.path.normpath(os.path.join(output_dir, argv.model_name))
 
-    fem = FrontEndManager()
-    func = read_model(orig_model_name + "_tmp.xml")
+    def clear_tmp_ir_files():
+        for suf in [".xml", ".bin", ".mapping"]:
+            # remove existing files
+            path_to_file = orig_model_name + "_tmp" + suf
+            if os.path.exists(path_to_file):
+                os.remove(path_to_file)
+
+    try:
+        prepare_emit_ir(graph=graph,
+                        data_type=graph.graph['cmd_params'].data_type,
+                        output_dir=argv.output_dir,
+                        output_model_name=argv.model_name,
+                        mean_data=mean_data,
+                        input_names=input_names,
+                        meta_info=non_default_params,
+                        use_temporary_path=True)
+    except Exception as err:
+        clear_tmp_ir_files()
+        raise Error('Exception occurred during serialization of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    finally:
+        # This graph cleanup is required to avoid double memory consumption
+        graph.clear()
+
+    try:
+        fem = FrontEndManager()
+        func = read_model(fem, orig_model_name + "_tmp.xml")
+    except Exception as err:
+        raise Error('Exception occurred while reading of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    finally:
+        clear_tmp_ir_files()
```

Comment:
please combine these two `try-except` blocks as well. Let us have only try and final and the exception is re-raised.

Reply:
done